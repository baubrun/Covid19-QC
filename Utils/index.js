import checkPropTypes from "check-prop-types";


export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test="${attr}"]`);
  return wrapper;
};

export const checkProps = (component, expected) => {
  const propsError = checkPropTypes(
    component.propTypes,
    expected,
    "props",
    component.name
  );
  return propsError
};
