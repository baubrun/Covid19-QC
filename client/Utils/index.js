import checkPropTypes from "check-prop-types";


export const checkProps = (component, expected) => {
  const propsError = checkPropTypes(
    component.propTypes,
    expected,
    "props",
    component.name
  );
  return propsError
};
