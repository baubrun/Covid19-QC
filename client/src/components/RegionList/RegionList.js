import React from "react";
import PropTypes from "prop-types";
import "./RegionList.css";


const RegionList = ({ regions, regionChange, loading }) => {
  return (
    <>
      {!loading && (
        <div id="list" 
        className="input-group">
          <select className="custom-select mt-3" onChange={regionChange}>
            <option value="Total">
              RÃ©gions
            </option>
            {regions.map((region, idx) => (
              <option key={idx} value={region}>
                {region}
              </option>
            ))}
          </select>
        </div>
      )}
    </>
  );
};

RegionList.propTypes = {
  regionChange: PropTypes.func,
  regions: PropTypes.array,
  loading: PropTypes.bool
};

export default RegionList;
