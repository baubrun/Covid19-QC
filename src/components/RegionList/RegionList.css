.custom-select {
    color: #235789;
    font-size: 2em !important;
}