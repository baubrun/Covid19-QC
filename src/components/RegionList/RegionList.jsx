import React from "react";

const RegionList = ({ data, regionChange }) => {
  return (
    <>
      <select className="custom-select"
      onChange={regionChange}
      >
        <option value="Total">Régions</option>
        {data.slice(0,-1).map((d, idx) => (
          <option key={idx} value={d.région}>
            {d.région}
          </option>
        ))}
      </select>
    </>
  );
};

export default RegionList;

