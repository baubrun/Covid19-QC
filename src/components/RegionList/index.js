export {default as RegionList} from "./RegionList"