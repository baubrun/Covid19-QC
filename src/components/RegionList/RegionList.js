import React from "react";
import PropTypes from "prop-types";
import "./RegionList.css";


const RegionList = ({ regions, regionChange, loading }) => {
  return (
    <>
      {!loading && (
        <div id="list" 
        className="input-group">
          <select className="custom-select mt-3" onChange={regionChange}>
            <option value="">
              RÃ©gions
            </option>
            {regions.map((region, idx) => (
              <option key={idx} value={region}>
                {region}
              </option>
            ))}
          </select>
        </div>
      )}
    </>
  );
};

RegionList.propTypes = {
  data: PropTypes.array,
  regionChange: PropTypes.func,
  loading: PropTypes.bool
};

export default RegionList;
