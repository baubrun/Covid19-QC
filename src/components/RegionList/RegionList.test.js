import React from "react";
import RegionList from "./RegionList";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { checkProps } from "../../../Utils";

const regions = [
  "Bas-Saint-Laurent",
  "Saguenay-Lac-Saint-Jean",
  "Capitale-Nationale",
];

const props = {
  regions,
  regionChange: () => {},
  loading: false,
};


describe("RegionList Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<RegionList {...props} />);
  });

  it("should match snapshot ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("Checking props", () => {
    it("should not throw warning", () => {

      const propsError = checkProps(wrapper, props);
      expect(propsError).toBeUndefined();
    });
  });

  describe("Matching Text", () => {
    console.log(wrapper);
  });
});
