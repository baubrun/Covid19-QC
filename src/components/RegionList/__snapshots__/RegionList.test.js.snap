// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionList Component should match snapshot  1`] = `
<Fragment>
  <div
    className="input-group"
    id="list"
  >
    <select
      className="custom-select mt-3"
      onChange={[Function]}
    >
      <option
        value=""
      >
        RÃ©gions
      </option>
      <option
        key="0"
        value="Bas-Saint-Laurent"
      >
        Bas-Saint-Laurent
      </option>
      <option
        key="1"
        value="Saguenay-Lac-Saint-Jean"
      >
        Saguenay-Lac-Saint-Jean
      </option>
      <option
        key="2"
        value="Capitale-Nationale"
      >
        Capitale-Nationale
      </option>
    </select>
  </div>
</Fragment>
`;
